mcs/class/System/System.Net/HttpListener.Mono.cs

```diff
@@ -100,12 +100,9 @@ internal SslStream CreateSslStream (Stream innerStream, bool ownsStream, RemoteC
 			lock (_internalLock) {
 				if (tlsProvider == null)
 					tlsProvider = MonoTlsProviderFactory.GetProvider ();
-				if (tlsSettings == null)
-					tlsSettings = MonoTlsSettings.CopyDefaultSettings ();
-				if (tlsSettings.RemoteCertificateValidationCallback == null)
-					tlsSettings.RemoteCertificateValidationCallback = MNS.Private.CallbackHelpers.PublicToMono (callback);
-				var sslStream = tlsProvider.CreateSslStream (innerStream, ownsStream, tlsSettings);
-				return sslStream.SslStream;
+					var settings = (tlsSettings ?? MonoTlsSettings.DefaultSettings).Clone ();
+					settings.RemoteCertificateValidationCallback = MNS.Private.CallbackHelpers.PublicToMono (callback);
```

Comment:
the indentation makes it look like this is supposed to be under the `if (tlsProvider == null)`, are you missing braces?